File: WatchBox/Shared/Models/MediaResult.swift
//
//  MediaResult.swift
//  SceneBox
//
//  Created by SpontaneousArray on 29.07.26.
//

import Foundation

nonisolated public struct MediaResult: Identifiable, Sendable {
    public let id: String          // IMDb id, e.g. "tt1375666"
    public let type: MediaType
    public let name: String
    public let year: String?
    public let posterURL: URL?
    public let description: String?
}

extension MediaResult: Hashable {
    public static func == (lhs: MediaResult, rhs: MediaResult) -> Bool { lhs.id == rhs.id }
    public func hash(into hasher: inout Hasher) { hasher.combine(id) }
}

File: lib/core/models/media_result.dart
import 'media_type.dart';

class MediaResult {
  const MediaResult({
    required this.id,
    required this.type,
    required this.name,
    this.year,
    this.posterUrl,
    this.description,
  });

  final String id;
  final MediaType type;
  final String name;
  final String? year;
  final Uri? posterUrl;
  final String? description;

  @override
  bool operator ==(Object other) =>
      identical(this, other) || other is MediaResult && id == other.id;

  @override
  int get hashCode => id.hashCode;
}
